openverse_catalog/dags/providers/provider_api_scripts/phylopic.py

```diff
@@ -13,272 +13,238 @@
+        """
+        Override for upstream "return True". Dated runs will only ever make 1 query so
+        they should not continue to loop.
+        """
+        return not bool(self.date)
+
+    @staticmethod
+    def _get_response_data(response_json) -> dict | list | None:
+        """
+        Intermediate method for pulling out results from a Phylopic API request.
+        """
+        if response_json and response_json.get("success") is True:
+            return response_json.get("result")
+
+    def get_batch_data(self, response_json):
+        """
+        Process the returned IDs.
+
+        The Phylopic API returns only lists of IDs in the initial request. We must take
+        this request and iterate through all the IDs to get the metadata for each one.
+        """
+        data = self._get_response_data(response_json)
+
+        if not data:
+            logger.warning("No content available!")
+            return None
+
+        return data
+
+    @staticmethod
+    def _image_url(uid: str) -> str:
+        return f"{PhylopicDataIngester.host}/image/{uid}"
+
+    @staticmethod
+    def _get_image_info(
+        result: dict, uid: str
+    ) -> tuple[str | None, int | None, int | None]:
+        img_url = None
+        width = None
+        height = None
+
+        image_info = result.get("pngFiles")
+        if image_info:
+            images = list(
+                filter(lambda x: (int(str(x.get("width", "0"))) >= 257), image_info)
+            )
+            if images:
+                image = sorted(images, key=lambda x: x["width"], reverse=True)[0]
+                img_url = image.get("url")
+                if not img_url:
+                    logging.warning(
+                        "Image not detected in url: "
+                        f"{PhylopicDataIngester._image_url(uid)}"
+                    )
+                else:
+                    img_url = f"{PhylopicDataIngester.host}{img_url}"
+                    width = image.get("width")
+                    height = image.get("height")
+
+        return img_url, width, height
+
+    @staticmethod
+    def _get_taxa_details(result: dict) -> tuple[list[str] | None, str]:
+        taxa = result.get("taxa", [])
+        taxa_list = None
+        title = ""
+        if taxa:
+            taxa = [
+                _.get("canonicalName")
+                for _ in taxa
+                if _.get("canonicalName") is not None
+            ]
+            taxa_list = [_.get("string", "") for _ in taxa]
+
+        if taxa_list:
+            title = taxa_list[0]
+
+        return taxa_list, title
+
+    @staticmethod
+    def _get_creator_details(result: dict) -> tuple[str | None, str | None, str | None]:
+        credit_line = None
+        pub_date = None
+        creator = None
+        submitter = result.get("submitter", {})
+        first_name = submitter.get("firstName")
+        last_name = submitter.get("lastName")
+        if first_name and last_name:
```

Comment:
I think it would be nice to add the submitter if there is a first_name OR last_name. Does that ever happen?